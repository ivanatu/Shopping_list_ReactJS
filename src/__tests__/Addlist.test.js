import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ReactDOM from 'react-dom' 
import renderer from 'react-test-renderer'
import Addlist from '../components/addlist'
import moxios from 'moxios';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import '../setupTests';


describe('<Addlist/>', () => {
    const wrapper = mount( <Addlist/> );

    it('has a valid snapshot', () => {
        const component = renderer.create(
        <Addlist />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
   
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div id="myModal"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-dialog"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-content"/>))
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-header"/>))
    });
    it('should render a button', () =>{
        expect(wrapper.find('button').length).toEqual(2)
    });
    it('should render <div> without throwing an error', () => {
        expect(wrapper.exists(<div className="modal-body"/>))
    });
    it('input should respond to change event and change the state', () => {
        wrapper.find('#list').simulate('change', { target: { name: 'list', value: 'mangoes' } });
        expect(wrapper.state().name).toEqual('mangoes')
    });
    it('render the inputs', () =>{
        expect(wrapper.find('form').length).toEqual(1);
        expect(wrapper.find('input').length).toEqual(1)
    });
    it('renders the add list form and submits data', () =>{
        wrapper.setState({list:'groceries'});
        wrapper.find("#login_form").simulate('submit', {preventDefault(){}});
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
