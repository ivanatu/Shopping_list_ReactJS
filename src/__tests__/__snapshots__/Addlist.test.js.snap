// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Addlist/> has a valid snapshot 1`] = `
<div>
  <div
    className="modal"
    id="myModal"
    role="dialog"
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h4>
            Add List
          </h4>
          <button
            className="close"
            data-dismiss="modal"
            id="close2"
            type="button"
          >
            Ã—
          </button>
        </div>
        <div
          className="modal-body"
        >
          <form
            className="register-form"
            id="login_form"
            method="POST"
            name="login_form"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              SHOPPING LIST NAME:
              <input
                className="form-control"
                id="list"
                name="list"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <button
              className="btn btn-info "
              type="submit"
            >
              Save List
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
